utility/WiFiStream.h

```diff
@@ -59,19 +60,37 @@ class WiFiStream : public Stream
 public:
   WiFiStream() {};
 
+#ifndef ESP8266
   // allows another way to configure a static IP before begin is called
   inline void config(IPAddress local_ip)
   {
     _local_ip = local_ip;
     WiFi.config( local_ip );
   }
+#endif
+
+  // allows another way to configure a static IP before begin is called
+  inline void config(IPAddress local_ip, IPAddress gateway, IPAddress subnet)
+  {
+    _local_ip = local_ip;
+    _gateway = gateway;
+    _subnet = subnet;
+#ifdef ESP8266
+    WiFi.config( local_ip, gateway, subnet );
+#else
+    WiFi.config( local_ip, IPAddress(0, 0, 0, 0), gateway, subnet );
```

Comment:
See my comment above to SUBNET_MASK and GATEWAY_IP_ADDRESS. I would prefer to always define them and to remove the variant with only a local IP but this needs testing and I don't have an Arduino with a WiFi shield.